sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/TriggerPartitionContext.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Azure.Messaging.EventHubs.Consumer;
+
+namespace Microsoft.Azure.WebJobs.EventHubs
+{
+    /// <summary>
+    /// Represents an Event Hub partition, its relative state, and its trigger state.
+    /// </summary>
+    public class TriggerPartitionContext : PartitionContext
+    {
+        /// <summary>
+        /// Determines whether this partition will checkpoint after this invocation succeeds.
+        /// </summary>
+        public bool IsCheckpointingAfterInvocation { get; internal set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TriggerPartitionContext"/> class.
+        /// </summary>
+        /// <param name="fullyQualifiedNamespace">The fully qualified Event Hubs namespace this context is associated with.</param>
+        /// <param name="eventHubName">The name of the Event Hub partition this context is associated with.</param>
+        /// <param name="consumerGroup">The name of the consumer group this context is associated with.</param>
+        /// <param name="partitionId">The identifier of the Event Hub partition this context is associated with.</param>
+        public TriggerPartitionContext(string fullyQualifiedNamespace, string eventHubName, string consumerGroup, string partitionId)
+            : base(fullyQualifiedNamespace ?? "default", eventHubName ?? "default", consumerGroup ?? "default", partitionId)
```

Comment:
I don't know that we want to default missing values - they should be populated.   Were there scenarios you ran into that these were `null`?

Reply:
The issue is a combination of existing test coverage and mocking strategy.

With the changes proposed, the following tests were then failing:
- [ProcessEvents_SingleDispatch_CheckpointsCorrectly](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubListenerTests.cs#L26-L58)
- [ProcessEvents_MultipleDispatch_CheckpointsCorrectly](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubListenerTests.cs#L60-L91)
- [ProcessEvents_Failure_Checkpoints](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubListenerTests.cs#L98-L134)
- [ProcessEvents_CancellationToken_CancelsExecution](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubListenerTests.cs#L273-L302)

These test were then failing because they cover [ProcessEventsAsync()](https://github.com/Azure/azure-sdk-for-net/blob/d0885934d8a34a7cbccd24f9fade404737f4bf65/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.cs#L151-L215) which now calls the new [UpdateCheckpointContext()](https://github.com/Azure/azure-sdk-for-net/blob/d0885934d8a34a7cbccd24f9fade404737f4bf65/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.cs#L217-L238) which in turn references (thereby instantiates) the [PartitionContext](https://github.com/Azure/azure-sdk-for-net/blob/d0885934d8a34a7cbccd24f9fade404737f4bf65/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHostPartition.cs#L25) (which wasn't test covered before) thereby now triggering it's instantiating during the unit test.

As each of the above tests mock the [EventProcessorHost](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubListenerTests.cs#L39-L44), but then Moq is unable to setup the properties passed to the partition context in the constructor of the [EventProcessorHostPartition](https://github.com/Azure/azure-sdk-for-net/blob/d0885934d8a34a7cbccd24f9fade404737f4bf65/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHostPartition.cs#L63-L67) (not new code) as they are not interceptable by the dyanmic proxy, null references are passed (in the test only) and are not accepted.

There are many solutions to this, but most are non-trivial changes to the mocking strategy / extracting interfaces etc., and guidance on the preferred way to balance the need to keep the existing test coverage but implement the new feature without sweeping changes to the mocks would be appreciated.